src/functions.php

```diff
@@ -231,6 +232,19 @@ function ($reason, $idx, Promise $aggregate) {
         ksort($results);
         return $results;
     });
+
+    if (true === $recursive) {
+        $promise = $promise->then(function ($results) use ($recursive, &$promises) {
+            foreach ($promises AS $promise) {
+                if (\GuzzleHttp\Promise\PromiseInterface::PENDING === $promise->getState()) {
+                    return all($promises, $recursive);
```

Comment:
This feeds promises into an EachPromise, even if they're fulfilled, right? Won't that invoke the `fulfilled` / `rejected` callbacks of the already settled promises potentially multiple times?